plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelTracingTelemetry.java

```diff
@@ -63,9 +64,22 @@ private Span createOtelSpan(SpanCreationContext spanCreationContext, Span parent
             OTelSpanKindConverter.convert(spanCreationContext.getSpanKind())
         );
         Span newSpan = new OTelSpan(spanCreationContext.getSpanName(), otelSpan, parentSpan);
+
+        // This attribute is added for inferred sampling
+        addInferredAttribute(newSpan, spanCreationContext);
+
         return newSpan;
     }
 
+    private void addInferredAttribute(Span newSpan, SpanCreationContext spanCreationContext) {
```

Comment:
Try to use the ReadableSpan instead of Baggage and the metadata.

Reply:
I will check on this as propagated span is also created, using Readable span might not work in that case. Propagated span are normal span only